---
title: "Assignment 4"
output: html_document
---
__Name:__ Chad Pittman

__Student ID:__ 914960676

## Assignment 4: Rice SNPs and GWAS

Remember to include the relevant code from the lab page here so that this file will knit.

**Exercise 0** Be sure that your final knitted file has good formatiing.  Make sure that you are using informative variable names.

**Exercise 1**

Now, create a data subset that contains a random sample of 5000 SNPs from the full data set.  Place the smaller data set in an object called `data.geno.5000`.  __Very important: you want to keep the first column, the one with the variety IDs, and you want it to be the first column.  AND You do not want it to show up randomly later on in the data set.__  Think about how to achieve this.

**EXERCISE 2:** Is there any evidence for populations structure (different sub populations)?  If so, how many sub populations do you think the MDS plot reveals?  What do you make of the individuals that are between the major groups? (Include your plot and code)

**EXERCISE 3:** 
* Use the `read_csv()` `head()` and `summary()` functions that you learned earlier to import and look at this file.  Import the file into an object called "data.pheno".
* Use a `join` function to merge the MDS scaled genotype data with the phenotype data. Use summary and head to look at the new object and make sure that it is as you expect.
* Include your code in the .Rmd


**EXERCISE 4:** Prepare three different plots to explore if subgroups vary by 1) Amylose content; 2) Pericarp color; 3) Region.  Do any of these seem to be associated with the different population groups? Briefly discuss.

**EXERCISE 5:** First, use a join function to combine the MDS data with the population assignments and place the result in `geno.mds.  Then re plot the MDS data, but include the population assignment in an informative way.  How do the populations assignments relate to the MDS plot? (Include your code and plot)

_The remaining exercises are from the GWAS lab_

**Exercise 6:** 
* Plot your chosen trait data
  * as a **single histogram** for all of the data
  * as **separate histograms** for each of the 4 population assignments made by FastStructure
  * as a **boxplot** separated by population. 
* Based on these histograms do you think that your trait varies by population? 
* **optional** Try using the "violin" geom.  
(Include code and plots)

**Exercise 7:** 
* Obtain the mean of your trait for each of the 4 fastStructure populations.
* Perform an ANOVA for your trait to test if it varies significantly by population.  Show your code, the ANOVA output, and provide an interpretation.
* Discuss: Do your results suggest a problem for GWAS?

**Exercise 8:** Describe and discuss the differences in the analysis with and without population structure correction.  Which do you think is the better one to follow-up on, and why?

**Exercise 9:** Look for genes close to your SNP at the [rice genome browser](http://rice.plantbiology.msu.edu/cgi-bin/gbrowse/rice/).  Pick a significant SNP from your analysis and enter its chromosome and position in the search box.  The browser wants you to enter a start and stop position, so for example, you should enter "Chr3:30449857..30449857" and then choose "show 20kb" from the pulldown menu on the left hand side.  Report the SNP you chose and the three closest genes.  These are candidate genes for determining the phenotype of your trait of interest in the rice population.  Briefly discuss these genes as possible candidates for the GWAS peak.  __Include a Screenshot of the genome browser in your answer__
